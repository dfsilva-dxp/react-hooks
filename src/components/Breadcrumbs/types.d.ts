export interface IBreadcrumbsProps {
  pageName?: string;
}
