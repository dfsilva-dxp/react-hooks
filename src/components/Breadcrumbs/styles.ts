import styled, { css } from "styled-components";

export const BreadcrumbsWrapper = styled.span`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: flex-start;
    gap: ${theme.spacings.xxsmall};
    padding: ${theme.spacings.small} 0;

    a {
      border-radius: ${theme.border.radius};
      &:hover,
      &:focus {
        outline: 2px solid ${theme.colors.green};
        outline-offset: ${theme.border.outlineOffiset};
      }
    }
    span {
      font-weight: ${theme.font.weight.bold};
      font-size: ${theme.font.sizes.small};
    }
    span:first-child {
      color: ${theme.colors.green};
    }
    svg {
      line-height: 0;
    }
  `}
`;

export const Disabled = styled.span`
  ${({ theme }) => css`
    cursor: not-allowed;
    color: ${theme.colors.label};
  `}
`;
