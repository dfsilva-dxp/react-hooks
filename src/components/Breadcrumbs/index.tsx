import { Link } from "react-router-dom";
import { CaretRight } from "phosphor-react";

import * as S from "./styles";

import { IBreadcrumbsProps } from "./types";

export default function Breadcrumbs({ pageName }: IBreadcrumbsProps) {
  return (
    <S.BreadcrumbsWrapper>
      <Link to="/">
        <span>Home</span>
      </Link>

      {pageName && (
        <>
          <CaretRight size={16} />
          <S.Disabled>{pageName}</S.Disabled>
        </>
      )}
    </S.BreadcrumbsWrapper>
  );
}
