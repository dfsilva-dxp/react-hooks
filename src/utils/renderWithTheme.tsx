import { ReactNode } from "react";
import { render, RenderResult } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { MyTheme } from "../styles";

export const renderWithTheme = (children: ReactNode): RenderResult =>
  render(<ThemeProvider theme={MyTheme}>{children}</ThemeProvider>);
