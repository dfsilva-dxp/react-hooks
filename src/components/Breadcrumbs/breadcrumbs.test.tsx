import { screen } from "@testing-library/react";

import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Breadcrumbs from ".";

const props = {
  pageName: "useState Hooks"
};

describe("<Breadcrumb/>", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<Breadcrumbs {...props} />);

    expect(screen.getByRole("link", { name: /home/i })).toHaveAttribute(
      "href",
      "/"
    );

    expect(screen.getByText(/usestate hooks/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
