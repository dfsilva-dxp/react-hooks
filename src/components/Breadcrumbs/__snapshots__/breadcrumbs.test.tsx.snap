// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb/> should render currectly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 0.5rem;
  padding: 1.5rem 0;
}

.c0 a {
  border-radius: 0.25rem;
}

.c0 a:hover,
.c0 a:focus {
  outline: 2px solid #00B37E;
  outline-offset: 0.125rem;
}

.c0 span {
  font-weight: 700;
  font-size: 0.875rem;
}

.c0 span:first-child {
  color: #00B37E;
}

.c0 svg {
  line-height: 0;
}

.c1 {
  cursor: not-allowed;
  color: #8D8D99;
}

<div>
  <span
    class="c0"
  >
    <a
      href="/"
    >
      <span>
        Home
      </span>
    </a>
    <svg
      fill="currentColor"
      height="16"
      viewBox="0 0 256 256"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <polyline
        fill="none"
        points="96 48 176 128 96 208"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
    </svg>
    <span
      class="c1"
    >
      useState Hooks
    </span>
  </span>
</div>
`;
